### PR TITLE
cleanup network stack as well as storage when container shuts down.

### PR DESCRIPTION
**Pull request by rhatdan**
This patch will cleanup the network stack when the container exits

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: libpod/container_api.go [hunk 1]
```diff
@@ -689,15 +689,22 @@ func (c *Container) Wait() (int32, error) {
...
 	if !c.locked {
 		c.lock.Lock()
 		defer c.lock.Unlock()
 		if err := c.syncContainer(); err != nil {
 			return err
 		}
 	}
+
+	// Stop the container's network namespace (if it has one)
+	if err := c.runtime.teardownNetNS(c); err != nil {
```
Comment (mheon) [inline libpod/container_api.go]: Need to also clear a few things in the DB after this - c.state.NetNS, c.state.IPAddress, c.state.SubnetMask. And probably need to do a c.save() afterwards to get that into the DB.

Comment (rhatdan) [reply libpod/container_api.go]: Fixed

**Comment by rhatdan** [comment]
@baude @mheon @umohnani8 @TomSweeneyRedHat PTAL

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by baude** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@rh-atomic-bot r=baude

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@rh-atomic-bot r=baude

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Comment by mheon** [comment]
Homu broken, tests green, going to manually merge.

**Pull request merged**